File: leetcode-submissions/3258_Count_Substrings_That_Satisfy_K-Constraint_I.ts
function countKConstraintSubstrings(s: string, k: number): number {
    let i :number =0 , j:number =0, count0:number  =0 , count1:number =0;
    const n : number =s.length;
    let ans :number =0;
    while(j<n){
        if(s[j]=="0"){
            count0++;
        }else{
            count1++;
        }
        while(count0>k && count1>k){
            if(s[i]=="0"){
            count0--;
            }else{
                count1--;
            }
            i++;

        }
        ans+=(j-i+1);
        j++;

    } 
    return ans ;
};

File: leetcode-submissions/3258_Count_Substrings_That_Satisfy_K-Constraint_I.cpp
class Solution {
public:
    int countKConstraintSubstrings(string s, int k) {
        // brute force 
        int ans =0;
        int n = s.length();
        for(int i=0;i<n;i++){
            int count0 =0 ;
            int count1= 0 ;

            for(int j=i;j<n;j++){
                if(s[j] == '0') {
                    count0++;
                } else {
                    count1++;
                }
               if (count1 <= k || count0 <= k) {
                    ans++;
                } else {
                    break; // No need to check further, this substring and beyond are invalid
                }

            }
        }
        return ans;
    }
};